packages/widgets/Source/Viewer/Viewer.js

```diff
@@ -2153,7 +2153,13 @@ function zoomToOrFly(that, zoomTarget, options, isFlight) {
 
       if (defined(zoomTarget.imageryProvider)) {
         // This is here for backward compatibility. It can be removed when readyPromise is removed.
-        rectanglePromise = zoomTarget.imageryProvider._readyPromise.then(() => {
+        let promise = Promise.resolve();
+        if (defined(zoomTarget.imageryProvider)) {
+          promise = zoomTarget.imageryProvider._readyPromise;
+        } else if (defined(zoomTarget.imageryProvider.readyPromise)) {
```

Comment:
The `else if` condition will never be met, if `zoomTarget.imageryProvider` is not defined.
Should the `if` condition instead be `defined(zoomTarget.imageryProvider._readyPromise)`?

Reply:
Good catch @jjhembd, that's an error on my part. It should now be fixed.